src/zap/zapimage.cpp

```diff
@@ -3617,7 +3619,24 @@ ZapNode * ZapImage::GetHelperThunk(CorInfoHelpFunc ftnNum)
 
     if (pHelperThunk == NULL)
     {
-        pHelperThunk = new (GetHeap()) ZapHelperThunk(ftnNum);
+        DWORD dwHelper = ftnNum;
+        switch (ftnNum)
+        {
+        case CORINFO_HELP_EE_VTABLE_FIXUP:
+        case CORINFO_HELP_EE_PERSONALITY_ROUTINE:
+        case CORINFO_HELP_EE_PERSONALITY_ROUTINE_FILTER_FUNCLET:
+        case CORINFO_HELP_STRCNS:
```

Comment:
What is special about `CORINFO_HELP_STRCNS` that it is here?

Reply:
This is a special case in which the helper is created when the jit requests a CORINFO_HELP_STRCNS_CURRENT_MODULE helper. getHelperFtn creates a ZapLazyHelperThunk, which sets up the module argument and contains a call to CORINFO_HELP_STRCNS. With my change, this becomes an indirect call.